client/product.go

```diff
@@ -552,3 +557,19 @@ func (c *ThreeScaleClient) DeployProductProxy(productID int64) (*ProxyJSON, erro
 	err = handleJsonResp(resp, http.StatusCreated, item)
 	return item, err
 }
+
+func (c *ThreeScaleClient) validateMappingRulesDuplication(productID int64, params Params) (bool, error) {
+	existingList, err := c.ListProductMappingRules(productID)
```

Comment:
On the create primitive, you need to list all the mapping rules first? Does not sound right. 

Maybe the caller can do that, or maybe has already done that, i.e. list all the product mapping rules.